### DISCUSSION TITLE
trying to get wayfire to work with nvidia drivers

### DISCUSSION THREAD

**Discussion by mrsteve0924** [WayfireWM/wayfire#1930] [General]
wayfire keeps crashing using nvidia driver 535.113.01-1 in archlinux with kernel 6.5.4

I have KMS enabled and added these to my env   GBM_BACKEND=nvidia-drm  & _GLX_VENDOR_LIBRARY_NAME=nvidia

not sure what else I should be doing to get this working.   here is a copy of the errors from my log.  https://dpaste.com//A95WXVGRD

looks like relevant errors are:

EE 01-10-23 21:14:04.752 - [GLES2] GL_INVALID_OPERATION error generated. EGLImage not supported
EE 01-10-23 21:14:04.753 - [render/gles2/renderer.c:139] Failed to create FBO
EE 01-10-23 21:14:04.753 - [GLES2] GL_INVALID_OPERATION error generated. EGLImage not supported
EE 01-10-23 21:14:04.753 - [render/gles2/renderer.c:139] Failed to create FBO
EE 01-10-23 21:14:04.753 - [GLES2] GL_INVALID_OPERATION error generated. EGLImage not supported

appreciate any help. thanks.

**Comment by soreau** [comment]
This is almost definitely a driver bug. You can try 1) a different version of the nvidia driver 2) nouveau or 3) the zink driver from mesa, on top of the mesa NVK driver or the proprietary vulkan driver.

**Comment by mrsteve0924** [comment]
yes nouveau i've been using since i started with wayfire and works fine.  i tried downgrading the nvidia driver and kernel but then had trouble even getting it to boot to the tty.  kept dropping me into emergency mode. said it couldn't mount my /boot. had to add vfat module to my initrd image to get it mount.  very strange, never had to do that before and not sure why i have to now.  i'm gonna have to work out that issue first before i can try starting wayfire.  it does boot to the tty with the latest kernel and latest nvidia driver without issue. 

not sure i 'm going to spend the time troubleshooting.  i really dont need to use the nvidia drivers. i was just curious if i can get them working.

**Comment by soreau** [comment]
> yes nouveau i've been using since i started with wayfire and works fine. i tried downgrading the nvidia driver and kernel but then had trouble even getting it to boot to the tty. kept dropping me into emergency mode. said it couldn't mount my /boot. had to add vfat module to my initrd image to get it mount. very strange, never had to do that before and not sure why i have to now. i'm gonna have to work out that issue first before i can try starting wayfire. it does boot to the tty with the latest kernel and latest nvidia driver without issue.
> 
> not sure i 'm going to spend the time troubleshooting. i really dont need to use the nvidia drivers. i was just curious if i can get them working.

Whether or not the nvidia drivers work for you is between you and nvidia. Closing this discussion for now.

**Comment by ammen99** [comment]
The biggest "problem" is that most of the low-level bits depend on wlroots, and wlroots doesn't officially support proprietary drivers, so getting support for this is hard :/ Even though it used to work awhile ago, so probably something broke in nvidia's drivers.

**Comment by soreau** [comment]
Ignoring the title, I would recommend using a intel integrated or radeon gpu for the best wayland experience, currently.

**Reply by mrsteve0924** [reply]
that does seem like the best path if I don't want to rely on the nouveau driver.  I gave my radeon rx 6700xt i bought last year to my son and took his 1660ti.  i think he is going to give me  a hard time swapping back . ha
